src/Util.php

```diff
@@ -75,4 +75,71 @@ public static function getTimestamp() {
         $micro = sprintf("%06d", ($t - floor($t)) * 1000000);
         return date('Y-m-d\TH:i:s.' . $micro . 'O', $t);
     }
+
+    /**
+     * determine which language out of an available set the user prefers most
+     *
+     * @method preferedLanguage returns the client's preferred language from a list of options.
+     * @param $available_languages {Array} list of language-tag-strings (must be lowercase) that are available
+     * @param $http_accept_language {String} a HTTP_ACCEPT_LANGUAGE string
+     *   (read from $_SERVER['HTTP_ACCEPT_LANGUAGE'] if left out)
+    */
+    public function preferedLanguage($available_languages, $http_accept_language = "auto")
+    {
+        // if $http_accept_language was left out, read it from the HTTP-Header
+        if ($http_accept_language == "auto") {
+            $http_accept_language = isset($_SERVER['HTTP_ACCEPT_LANGUAGE']) ? $_SERVER['HTTP_ACCEPT_LANGUAGE'] : '';
+        }
+
+        preg_match_all(
+            "/([[:alpha:]]{1,8})(-([[:alpha:]|-]{1,8}))?" .
+            "(\s*;\s*q\s*=\s*(1\.0{0,3}|0\.\d{0,3}))?\s*(,|$)/i",
+            $http_accept_language,
+            $hits,
+            PREG_SET_ORDER
+        );
+
+        // default language (in case of no hits) is the first in the array
+        $bestlang = $available_languages[0];
```

Comment:
You should better check first that `$available_languages` is not an empty array.